ecl/hqlcpp/hqlsource.cpp

```diff
@@ -5138,6 +5160,27 @@ IHqlExpression * MonitorExtractor::isKeyableFilter(IHqlExpression * left, IHqlEx
             reason.set(KFRnokey);
         return NULL;
 
+    case no_substring:
+        {
+            IHqlExpression * range = left->queryChild(1);
+            if (range->getOperator() == no_rangeto)
+            {
+                IValue *end = range->queryChild(0)->queryValue();
+                if (!end)
+                    break;
```

Comment:
minor: This shouldn't be a break, but should fall through to the statement on line 5181.  The difference is in the reason given in the error message if the expression is marked as keyed and cannot be performed.